### PR TITLE
chore: Upgrade to BlueprintJS 5 - SCE-4

### PR DESCRIPTION
**Pull request by kaloster**
JIRA Issue [SCE-4](https://czi.atlassian.net/browse/SCE-4)

Changes:

- Upgrade to BlueprintJS 5
- Removed popover import - now part of core
- Fixed breaking tests
- Updated snapshots
- General namespace updated to bp5

rdev -> [HERE](https://pumped-woodcock.dev-sc.dev.czi.team/d/super-cool-spatial.cxg/?spatial=true)

<details open>
  <summary>Argus Stack Details</summary>
  <br />
  <table>
    <tr>
      <th>Name</th>
      <td>kaloster-update-blueprintjs</td>
    </tr>
    <tr>
      <th>Short Name</th>
      <td>pumped-woodcock</td>
    </tr>
    <tr>
      <th>Base URL</th>
      <td><a href="https://pumped-woodcock.dev-sc.dev.czi.team" target="_blank">https://pumped-woodcock.dev-sc.dev.czi.team</a></td>
    </tr>
  </table>
</details>

---

[SCE-4]: https://czi.atlassian.net/browse/SCE-4?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: client/__tests__/e2e/e2e.test.ts-snapshots/dataset-pbmc3k-cxg-graph-instance-layout-gra-26a66-taset-and-collection-from-breadcrumbs-appears-2-chromium-linux.txt [hunk 1]
```diff
@@ -1 +1 @@
-<a class="bp3-breadcrumb datasetSelector__datasetSelectorBreadcrumb___Bjpl3" tabindex="0">Lorem ipsum dolor sit amet</a>
\ No newline at end of file
+<a class="bp4-breadcrumb datasetSelector__datasetSelectorBreadcrumb___Bjpl3" tabindex="0">Lorem ipsum dolor sit amet</a>
```
Comment (tihuan) [inline client/__tests__/e2e/e2e.test.ts-snapshots/dataset-pbmc3k-cxg-graph-instance-layout-gra-26a66-taset-and-collection-from-breadcrumbs-appears-2-chromium-linux.txt:1]: just curious, should this be bp5 instead 😄 ?

File: client/configuration/webpack/obsoleteBrowsers.js [hunk 2]
```diff
@@ -10,7 +10,7 @@ const regex = new RegExp(datasetRegex.slice(1, -1));
 if (dataset && !regex.test(navigator.userAgent)) {
   const root = document.getElementById("root");
   root.remove();
-  const portals = document.getElementsByClassName("bp3-portal");
+  const portals = document.getElementsByClassName("bp4-portal");
```
Comment (tihuan) [inline client/configuration/webpack/obsoleteBrowsers.js:13]: is there bp5 portal lol?

File: client/package.json [hunk 3]
```diff
@@ -54,10 +54,9 @@
   ],
   "dependencies": {
     "@aws-sdk/client-secrets-manager": "^3.13.0",
-    "@blueprintjs/core": "^3.44.0",
-    "@blueprintjs/icons": "^3.19.0",
-    "@blueprintjs/popover2": "^0.6.0",
-    "@blueprintjs/select": "^3.16.0",
+    "@blueprintjs/core": "^5.11.0",
+    "@blueprintjs/icons": "^5.11.0",
+    "@blueprintjs/select": "^5.2.2",
```
Comment (tihuan) [inline client/package.json:59]: thank you SO much for doing this upgrade!! 🏆  🚀 👏 !!

File: client/src/components/menubar/clip/index.tsx [hunk 4]
```diff
@@ -5,37 +5,28 @@ import {
...
   Tooltip,
 } from "@blueprintjs/core";
 import { IconNames } from "@blueprintjs/icons";
 
-import { tooltipHoverOpenDelay } from "../../globals";
+import { tooltipHoverOpenDelay } from "../../../globals";
 // @ts-expect-error ts-migrate(2307) FIXME: Cannot find module './menubar.css' or its correspo... Remove this comment to see the full error message
-import styles from "./menubar.css";
+import styles from "../menubar.css";
+import { ClipProps } from "./types";
```
Comment (tihuan) [inline client/src/components/menubar/clip/index.tsx:17]: thanks for adding the type here!!

File: client/src/components/menubar/clip/index.tsx [hunk 5]
```diff
@@ -66,12 +61,14 @@ const Clip = React.memo((props) => {
...
+              ref={tooltipRef}
+              {...tooltipProps}
             />
           </Tooltip>
-        }
+        )}
         position={Position.BOTTOM_RIGHT}
-        onOpening={handleClipOpening}
-        onClosing={handleClipClosing}
+        onOpening={() => handleClipOpening}
```
Comment (tihuan) [inline client/src/components/menubar/clip/index.tsx:70]: I assume `handleClipOpening` is already a function, why do we need to wrap it in an arrow function 💡 ?

File: client/src/components/menubar/clip/index.tsx [hunk 6]
```diff
@@ -145,6 +146,6 @@ const Clip = React.memo((props) => {
       />
     </ButtonGroup>
   );
-});
+}
 
-export default Clip;
+export default React.memo(Clip);
```
Comment (tihuan) [inline client/src/components/menubar/clip/index.tsx:151]: Nice memo!

**Review by tihuan — COMMENTED** [review]
Looks really good so far thanks so much! Just some minor questions 😄 

Last nit is that I saw some test snapshots still have bp3 and that there's maybe two or three code comments that still refer to `bp3` that we probably need to update to `bp5`? 

Thank you so much again for this two major version upgrade!

File: client/src/components/labelInput.tsx [hunk 7]
```diff
@@ -1,5 +1,5 @@
 import React from "react";
-import { InputGroup, MenuItem, Keys } from "@blueprintjs/core";
```
Comment (kaloster) [inline client/src/components/labelInput.tsx:2]: blueprintjs Keys is deprecated, using named keys instead

**Review by tihuan — COMMENTED** [review]
PR looks awesome thank you! Just wondering why there are so many deleted test snapshots when there's no test changes 😄 Thank you!

**Comment by tihuan** [comment]
@kaloster commented:

Hey Timmy! good question :slightly_smiling_face: I've updated all snapshots and removed all the ones that still had bp3 references after the update - could it be that those are just leftovers from previous runs, like snapshots that just remained over the changes to tests in the past year? most of the ones removed are from 6-8 months ago. In any case tests seem to pass after removing those snapshots

**Pull request merged**